language/diem-vm/src/diem_transaction_executor.rs

```diff
@@ -771,7 +773,7 @@ pub(crate) enum PreprocessedTransaction {
     WaypointWriteSet(WriteSetPayload),
     BlockMetadata(BlockMetadata),
     WriteSet(Box<SignatureCheckedTransaction>),
-    InvalidSignature,
+    Error(StatusCode),
```

Comment:
The error change should be fine since it is only used internally within the adapter, but as I commented separately, I hope we can avoid the need for this by moving the extra signature checking elsewhere.